app/views/glossary_terms/index.html.erb

```diff
@@ -0,0 +1,31 @@
+<%
+  @title = :glossary_term_index_title.t
+
+  tabs = [
+    link_to(:create_glossary_term.t, new_glossary_term_path)
+  ]
+  @tabsets = { right: draw_tab_set(tabs) }
+%>
+
+<div class="row">
+  <div class="col-xs-12 max-width-text">
+    <%= :glossary_term_index_intro.tp %>
+  </div>
+</div>
+
+<table class="table table-striped">
+  <% @glossary_terms.each do |term| %>
+    <tr>
+      <td class="max-width-text">
+        <h4><%= link_to(term.name, glossary_term_path(term.id)) %>:</h4>
+        <%= term.description.tpl %>
+      </td>
+      <td>
+        <%= destroy_button(term) if in_admin_mode? %>
+        <%= if term&.thumb_image_id&.nonzero?
+          thumbnail(term.thumb_image, votes: true)
+        end %>
+      </td>
+    </tr>
```

Comment:
I did notice there is some significant n+1 query behavior happening on this page.  It took over 1700 queries to render the page with a pattern of a query against images followed by a query against image_votes.  This could probably be solved using the `.includes` method off the Rails finder.  That's a way more important performance issue to address than worrying about the cost of adding a partial.

Reply:
Fixed in 4116a85445b0ab0ff5292b5ab4e01ca18c32ab17. 
Thanks mo-nathan for pointing out the problem and the solution.

The relevant code was copied from `master`, so we have the same problem there.
**Do you think it's worth adding a gem like Bullet to see if master has other n+1 problems?**

Reply:
I like bullet and would support adding it.  However, I don't love bullet.  It is very likely that it will generate a very large number of warnings many of which are simply not worth addressing and sifting through them takes time, but there will probably be a good number of easy wins like this one.  By way of warning, the suggestions that bullet makes sometimes are bad and cause even more work to be done so it cannot be used blindly.  Typically cases where it does this really require a deep refactor to make efficient.  I am quite confident that we have many N+1 query issues that bullet would find that would improve the performance of various pages in our site.  However, let's consider that only after we address all the stuff we already have on our plate.

Reply:
I wouldn't worry about N+1 too much in general.  I mean sure, keep it in mind and don't go out of your way to create them! :)  But only a few pages account for the vast majority of our performance problems -- show observation, show name, observation index, rss log index.  *Those* pages we should probably be checking up on from time to time.  Even just watching the logs on a dev VM is enough to spot obvious issues.  But glossary pages are really low-profile, so it's not a big deal in my opinion.  And the database tables this controller is dealing with are really small.  Anything dealing with joins between observations and images, for example, is a totally different story.

Reply:
I added a ticket re bullet. https://www.pivotaltracker.com/story/show/174609339